benchmark.go

```diff
@@ -39,8 +42,102 @@ import (
 var (
 	errNegativeDuration = errors.New("duration cannot be negative")
 	errNegativeMaxReqs  = errors.New("max requests cannot be negative")
+
+	// using a global _quantiles slice and _percentileMap mainly for ease of testing, and not passing the same array
+	// around to multiple functions
+	_quantiles     = []float64{0.5, 0.9, 0.95, 0.99, 0.999, 0.9995, 1.0}
+	_percentileMap = map[float64]string{
+		0.5:    "P50",
+		0.9:    "P90",
+		0.95:   "P95",
+		0.99:   "P99",
+		0.999:  "P999",
+		0.9995: "P9995",
+		1.0:    "P100",
+	}
```

Comment:
Since the map and the slice are closely related, I'd suggest having a single source of truth (probably the map), and generate the slice based on that.
E.g.,
```go
_percentileMap = map[float64]string{
  [...]
}
_quantiles = mapKeysPercentileMap(_percentileMap)

[.. later in the file ..]
func mapKeysPercentileMap(m map[float64]string) []float64 {
  keys := make([]float64, len(m))
  for k := range m {
    keys = append(keys, k)
  }
  sort.Float64s(keys)
  return keys
}
```

This avoids the data getting out of sync